### PR TITLE
feat(codegen): add text2sql views code generation

### PR DESCRIPTION
**Pull request by micpst**
This PR adds code generation for text2sql views and refactors autodiscovery for codegen.

### CODE REVIEW

File: src/dbally_cli/text2sql.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    Args:
+        file_path: The path to the file where the view will be generated.
+        db_url: The database connection string.
+        llm_object: The path to the LLM object.
+        llm_description: Generate a description using the LLM object.
+        similarity_index_factory: The path to the similarity index factory.
+    """
+    click.echo("Generating Text2SQL view...")
+
+    root, ext = os.path.splitext(file_path)
```
Comment (karllu3) [inline src/dbally_cli/text2sql.py:64]: better use pathlib it has been added in python 3.4

Comment (micpst) [reply src/dbally_cli/text2sql.py]: A matter of preference, it does not change the functionality.

**Comment by mhordynski** [comment]
Code looks great :1st_place_medal: Let's add documentation and we can merge it

**Comment by micpst** [comment]
I'll update the documentation in a separate PR for both code generation and freeform views.

**Review by mhordynski — APPROVED** [review]

**Pull request merged**